### ISSUE TITLE
jwt::decode() in apple provider? (with php-jwt >= 6.2.0)

### ISSUE THREAD

**Issue by dani3l3** [hybridauth/hybridauth#1331]
FIrebase-jwt 6.2.0 has changed the vay you call jwt::decode() with regards to how you pass in the key --> https://github.com/firebase/php-jwt
https://stackoverflow.com/questions/72278051/why-is-jwtdecode-returning-status-kid-empty-unable-to-lookup-corr

from 
use Firebase\JWT\JWT;
$token = JWT::decode($jwt, $secretKey, array('HS512'));

to
use Firebase\JWT\JWT;
use Firebase\JWT\Key;
$data = JWT::decode($token, new Key($topSecret, 'HS256');

I see the apple provider https://github.com/hybridauth/hybridauth/blob/master/src/Provider/Apple.php still seems to use the old syntax... 

...have you verified it still works and they haven't broken you? 

...I am afraid of upgrading my libraries thru composer...

**Comment by sputnik73** [comment]
Thank you for the information. You are right, the current code does not support the new version FIrebase-jwt 6.2.0. It seems it is not possible to restrict the version for suggested packages in composer.json, so I think both ways have to be implemented in the code and executed depending on the installed version of Firebase-jwt.

It seems the version can be checked using
`\Composer\InstalledVersions::getVersion('my/package')`

I will try to implement this during the next days. If somebody has a better approach, suggestions are welcome of course!

**Comment by dani3l3** [comment]
Thank you. I'll stay tuned for when you have your fix ready so then I can safely update on my end.

**Comment by dani3l3** [comment]
It's much appreciated: I understand everybody is busy and I don't blame you, I can hold back my changes till you are ready... it's Google who shouldn't just break stuff on everybody like that!

**Comment by sputnik73** [comment]
Sorry for the delay! I created a pull request and tested the code with firebase-jwt 6.2.0 and 5.5.1 successfully.

**Comment by dani3l3** [comment]
Of course.... but regardless on whether our/your code is "bad" or "good", forward-looking or not, firebase/google should have really NOT broken this: you don't just change such an important method like that with no announcement and then expect everybody else to run after your changes! They could have also provided an overload to keep supporting the old syntax! This behaviour shows very little respect for their clients, I find it extremely rude of them. Therefore, if they do hacks, we have to do hacks as well to run after their hacks.... what else can we do? ;-) 

PS - I asked also on https://github.com/hybridauth/hybridauth/pull/1337 but I see the latest release is still 3.8.0 (released in april). When can I expect to find a new version that contains this fix on packagist so that I can update thru composer?

**Comment by ApacheEx** [comment]
released as v3.8.1
